src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -59,6 +59,13 @@ public static function compile(Route $route)
         $staticPrefix = $result['staticPrefix'];
 
         $pathVariables = $result['variables'];
+
+        foreach ($pathVariables as $pathParam) {
+            if ('_fragment' == $pathParam) {
```

Comment:
Though not documented, it looks like the `variables` output by compilePattern will always be strings, so a strict type equality would not be required.

Unless of course this is a coding standard, or maybe if a test could show it was required (as it seems to me without proof either way then both strict and non-strict comparison have the possibility of causing a bug).